WordPress/gutenberg
# Site Editor: update index view for pages

Part of https://github.com/WordPress/gutenberg/issues/59659
Related https://github.com/WordPress/gutenberg/pull/59792 and https://github.com/WordPress/gutenberg/issues/55083

## What?

Makes the DataViews-powered Page the index view for the pages section.

https://github.com/WordPress/gutenberg/assets/583546/3c0a4e9e-e0da-4589-804e-f948a9b59de4

## Why?

It's the direction we want to go.

## How?

- Update sidebar area cc993b5dbfaa782109e8b41696878194f2c16ecf
- Delete old sidebar 3ac9349f7c420844c3278aba7104ff195bb0efc6
- Update content area 05af5c988a934a24903270261460f79f9a483e6a
- Update default layout to list a2a373a10be65add43b8d743524976f98e023866
- Update preview area bf846c460f8caa6479786e9230a28224c6d8d351
- Clear params for page path e5f7e7dbaf9798ddcea12ca7cb692504b0353f16

## Testing Instructions

Go to the Site Editor and visit the Pages section. It should display the list layout for pages.

## Follow-ups

I've ran into the following issues while testing. This also happens in `trunk` so I plan to investigate separately:

- The list layout (content&preview) don't use the URL params to update selection on first load.
- For custom views, the list layout doesn't show the preview.
- When it goes to the editor, it uses `path=page` instead of `postType=page`. The side-effect is that when coming back, it has the `path` and goes directly to list layout. It should go to the details view for the time being.
